boot/freeldr/freeldr/arch/realmode/fathelp.inc

```diff
@@ -61,12 +65,28 @@ start:
 
 
 FatHelperEntryPoint:
+    /* Clear the segment values of the indirect far calls to make sure it works */
+    xor bx, bx
+    mov word ptr ss:[bp-(ReadSectorsOffset-2)], bx
+    mov word ptr ss:[bp-(ReadClusterOffset-2)], bx
+    mov word ptr ss:[bp-(PutCharsOffset-2)], bx
+
     /* First save AX - the start cluster of freeldr.sys */
     push ax
 
+    /* Save ds */
+    push ds
+
+    /* Prepare ds before PutChars call */
+    mov bx, cs
+    mov ds, bx
```

Comment:
This is also something you do implicitly when calling `cs_writestr`.
Why is it now necessary to change the ds before calling these functions?

Reply:
i added the `CS_PutChars` function to resolve this issue.

Reply:
I'm asking _why_ is this necessary.

Reply:
> This is also something you do implicitly when calling `cs_writestr`. Why is it now necessary to change the ds before calling these functions?

since i can't use the `cs_writestr` function because its not in the helper code so that's why i changed the ds at that moment.